c7n/resources/cw.py

```diff
@@ -466,6 +468,87 @@ def get_dimensions(self, resource):
         return [{'Name': 'LogGroupName', 'Value': resource['logGroupName']}]
 
 
+@resources.register('log-metric-filter')
+class LogMetricFilterResource(QueryResourceManager):
+    class resource_type(TypeInfo):
+        service = 'logs'
+        enum_spec = ('describe_metric_filters', 'metricFilters', None)
+        arn = False
+        id = name = 'filterName'
+        date = 'creationTime'
+        cfn_type = 'AWS::Logs::MetricFilter'
+
+
+@LogMetricFilterResource.filter_registry.register('alarm')
+class MetricFilterAlarmFilter(ValueFilter):
+    """
+    Filter log metric filters based on associated alarms.
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: log-metric-filters-with-alarms
+            resource: aws.log-metric-filter
+            filters:
+              - type: alarm
+                key: AlarmName
+                value: present
+    """
+
+    schema = type_schema('alarm', rinherit=ValueFilter.schema)
+    annotation_key = 'c7n:MetricAlarms'
+    FetchThreshold = 10  # below this number of resources, fetch alarms individually
+
+    def augment(self, resources):
+        """Add alarm details to log metric filter resources
+
+        This includes all alarms where the metric name and namespace match
+        a log metric filter's metric transformation.
+        """
+
+        if len(resources) < self.FetchThreshold:
+            client = local_session(self.manager.session_factory).client('cloudwatch')
+            for r in resources:
+                r[self.annotation_key] = list(itertools.chain(*(
+                    self.manager.retry(
+                        client.describe_alarms_for_metric,
+                        Namespace=t['metricNamespace'],
+                        MetricName=t['metricName'])['MetricAlarms']
+                    for t in r.get('metricTransformations', ())
+                )))
+        else:
+            alarms = Alarm(self.manager.ctx, {}).resources()
```

Comment:
alarms = self.get_resource_manager('aws.alarm') is the preferred mechanism for getting another resource manager